packages/nx/src/command-line/migrate.ts

```diff
@@ -487,72 +488,99 @@ function versions(root: string, from: Record<string, string>) {
+  const migrationsCache: Record<string, ResolvedMigrationConfiguration> = {};
+  const resolvedVersionCache: Record<string, string> = {};
 
-  function fetchMigrations(
-    packageName,
-    packageVersion,
-    setCache: (packageName: string, packageVersion: string) => void
-  ): Promise<ResolvedMigrationConfiguration> {
-    const cacheKey = packageName + '-' + packageVersion;
-    return Promise.resolve(resolvedVersionCache[cacheKey])
-      .then((cachedResolvedVersion) => {
-        if (cachedResolvedVersion) {
-          return cachedResolvedVersion;
-        }
+  function getMigrationCache(packageName: string, packageVersion: string) {
+    return migrationsCache[`${packageName}-${packageVersion}`];
+  }
 
-        resolvedVersionCache[cacheKey] = resolvePackageVersionUsingRegistry(
-          packageName,
-          packageVersion
-        );
-        return resolvedVersionCache[cacheKey];
-      })
-      .then((resolvedVersion) => {
-        if (
-          resolvedVersion !== packageVersion &&
-          migrationsCache[`${packageName}-${resolvedVersion}`]
-        ) {
-          return migrationsCache[`${packageName}-${resolvedVersion}`];
-        }
-        setCache(packageName, resolvedVersion);
-        return getPackageMigrationsUsingRegistry(packageName, resolvedVersion);
-      })
-      .catch(() => {
-        logger.info(`Fetching ${packageName}@${packageVersion}`);
+  function setMigrationCache(
+    packageName: string,
+    packageVersion: string,
+    migrations: ResolvedMigrationConfiguration
+  ) {
+    migrationsCache[`${packageName}-${packageVersion}`] = migrations;
+  }
+
+  const packageRegistryCache: Record<string, string> = {};
+
+  const pm = detectPackageManager();
 
-        return getPackageMigrationsUsingInstall(packageName, packageVersion);
-      });
+  async function getRegistryOfPackage(packageName: string) {
+    const configKeyPrefix = packageName.startsWith('@')
+      ? `${packageName.substring(0, packageName.indexOf('/'))}:`
+      : '';
+
+    if (!packageRegistryCache[packageName]) {
+      const { stdout } = await execAsync(
+        `${pm} config get ${configKeyPrefix}registry`
+      );
+
+      packageRegistryCache[configKeyPrefix] = stdout.trim();
+    }
+    return packageRegistryCache[configKeyPrefix];
   }
 
-  return function nxMigrateFetcher(
+  async function fetchMigrations(
     packageName: string,
     packageVersion: string
   ): Promise<ResolvedMigrationConfiguration> {
-    if (migrationsCache[`${packageName}-${packageVersion}`]) {
-      return migrationsCache[`${packageName}-${packageVersion}`];
+    if (
+      (await getRegistryOfPackage(packageName)) ===
+      'https://npm.pkg.github.com/'
+    ) {
+      /**
+       * The github npm package registry does not yet support custom fields,
+       * meaning that `npm view` will not return the nx migration information,
+       * so we have to fallback to getting the information with a full install.
+       */
```

Comment:
nit: Move this above the if statement, s.t. the condition is closer to the logic